diff --git a/src/Makefile.am b/src/Makefile.am
index e9aacfe..c786059 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -322,7 +322,7 @@ libgnome_shell_libadd =		\
 	$(NULL)
 
 libgnome_shell_menu_la_LDFLAGS = $(libgnome_shell_ldflags)
-libgnome_shell_menu_la_LIBADD = $(GNOME_SHELL_LIBS)
+libgnome_shell_menu_la_LIBADD = $(GNOME_SHELL_LIBS) $(MUTTER_LIBS)
 libgnome_shell_menu_la_CPPFLAGS = $(GNOME_SHELL_CFLAGS)
 
 libgnome_shell_base_la_LIBADD = $(libgnome_shell_libadd)
